manifests/apache.pp

```diff
@@ -0,0 +1,154 @@
+      mode   => '0755',
+      purge  => true,
+    }
+
+    if $pulp::enable_http or $pulp::enable_puppet {
+      apache::vhost { 'pulp-http':
+        priority            => '05',
+        docroot             => '/srv/pulp',
+        port                => 80,
+        servername          => $::fqdn,
+        serveraliases       => [$::hostname],
+        additional_includes => '/etc/pulp/vhosts80/*.conf',
+      }
+    }
+
+    apache::vhost { 'pulp-https':
+      priority                   => '05',
+      docroot                    => '/srv/pulp',
+      port                       => 443,
+      servername                 => $::fqdn,
+      serveraliases              => [$::hostname],
+      ssl                        => true,
+      ssl_cert                   => $pulp::https_cert,
+      ssl_key                    => $pulp::https_key,
+      ssl_ca                     => $pulp::ca_cert,
+      ssl_verify_client          => 'optional',
+      ssl_protocol               => ' all -SSLv2',
+      ssl_options                => '+StdEnvVars +ExportCertData',
+      ssl_verify_depth           => '3',
+      wsgi_process_group         => 'pulp',
+      wsgi_application_group     => 'pulp',
+      wsgi_daemon_process        => 'pulp user=apache group=apache processes=1 threads=8 display-name=%{GROUP}',
+      wsgi_pass_authorization    => 'On',
+      wsgi_import_script         => '/srv/pulp/webservices.wsgi',
+      wsgi_import_script_options => {
+        'process-group'     => 'pulp',
+        'application-group' => 'pulp',
+      },
+      wsgi_script_aliases        => {
+        '/pulp/api' => '/srv/pulp/webservices.wsgi',
+      },
+      directories                => [
+        {
+          'path'     => 'webservices.wsgi',
+          'provider' => 'files',
+        },
+        {
+          'path'     => '/srv/pulp',
+          'provider' => 'directory',
+        },
+        {
+          'path'     => '/pulp/static',
+          'provider' => 'location',
+        },
+      ],
+      aliases                    => [{
+          alias           => '/pulp/static',
+          path            => '/var/lib/pulp/static',
+          options         => ['Indexes'],
+          custom_fragment => 'SSLRequireSSL'
+        }
+      ],
+      options                    => ['SymLinksIfOwnerMatch'],
+      custom_fragment            => '# allow older yum clients to connect, see bz 647828
+	  SSLInsecureRenegotiation on
+	  AddDefaultCharset UTF-8',
+    }
+
+    if $pulp::enable_rpm {
+      file { '/etc/httpd/conf.d/pulp_rpm.conf':
+        ensure  => file,
+        content => template('pulp/etc/httpd/conf.d/pulp_rpm.conf.erb'),
+        owner   => 'root',
+        group   => 'root',
+        mode    => '0644',
+      }
+
+      file { '/etc/pulp/vhosts80/rpm.conf':
+        ensure  => file,
+        content => template('pulp/vhosts80/rpm.conf'),
+        owner   => 'apache',
+        group   => 'apache',
+        mode    => '0600',
+      }
+    }
```

Comment:
These plugin definitions end up looking very similar, maybe we should make them a `define` for re-use?